tarodb/asset_minting_test.go

```diff
@@ -535,6 +633,13 @@ func TestCommitBatchChainActions(t *testing.T) {
 		assetBalance, ok := assetBalancesByGroup[groupKey]
 		require.True(t, ok)
 
+		// One asset was minted into an existing group, so the value
+		// of the group genesis asset must be deducted from the group
+		// balance before comparing to the minted asset.
+		if bytes.Equal(groupKey[:], existingGroupKey[:]) {
```

Comment:
I like that we were able to mostly re-use the existing test, but also feel like we should be able to have a more isolated test for this functionality as well. Eg: check that the call to fetch all the groups shows this asset as well. Still getting through the diff though, the itests could also be a natural place for this logic to live.

Reply:
Fair - I think almost all the new queries and logic changes are not used directly by the planter or caretaker, but as part of existing functions, so it felt harder to test them in isolation. Some of this may indeed overlap with itest though.